wandb/sdk/system/assets/network.py

```diff
@@ -0,0 +1,134 @@
+
+try:
+    import psutil
+except ImportError:
+    psutil = None
+
+from wandb.sdk.system.assets.aggregators import aggregate_mean
+from wandb.sdk.system.assets.asset_registry import asset_registry
+from wandb.sdk.system.assets.interfaces import (
+    Interface,
+    Metric,
+    MetricsMonitor,
+    MetricType,
+)
+
+if TYPE_CHECKING:
+    from typing import Deque
+
+    from wandb.sdk.internal.settings_static import SettingsStatic
+
+
+class NetworkSent:
+    """
+    Network bytes sent.
+    """
+
+    name = "network.sent"
+    metric_type: MetricType = "gauge"
+    samples: "Deque[float]"
+
+    def __init__(self) -> None:
+        self.samples = deque([])
+        self.sent_init = psutil.net_io_counters().bytes_sent
+
+    def sample(self) -> None:
+        self.samples.append(psutil.net_io_counters().bytes_sent - self.sent_init)
+
+    def clear(self) -> None:
+        self.samples.clear()
+
+    def aggregate(self) -> dict:
+        if not self.samples:
+            return {}
+        aggregate = aggregate_mean(self.samples)
+        # todo: this is an adapter for the legacy metrics system
+        # return {"network": {self.name: aggregate}}
+        return {self.name: aggregate}
+
+
+class NetworkRecv:
+    """
+    Network bytes received.
+    """
+
+    name = "network.recv"
+    metric_type: MetricType = "gauge"
+    samples: "Deque[float]"
+
+    def __init__(self) -> None:
+        self.samples = deque([])
+        self.recv_init = psutil.net_io_counters().bytes_recv
+
+    def sample(self) -> None:
+        self.samples.append(psutil.net_io_counters().bytes_recv - self.recv_init)
+
+    def clear(self) -> None:
+        self.samples.clear()
+
+    def aggregate(self) -> dict:
+        if not self.samples:
+            return {}
+        aggregate = aggregate_mean(self.samples)
+        # todo: this is an adapter for the legacy metrics system
+        # return {"network": {self.name: aggregate}}
+
+        return {self.name: aggregate}
+
+
+@asset_registry.register
+class Network:
+    def __init__(
+        self,
+        interface: "Interface",
+        settings: "SettingsStatic",
+        shutdown_event: mp.synchronize.Event,
+    ) -> None:
+        self.name = self.__class__.__name__.lower()
+        self.metrics: List[Metric] = [
+            NetworkSent(),
+            NetworkRecv(),
+        ]
+        self.metrics_monitor = MetricsMonitor(
+            self.metrics,
+            interface,
+            settings,
+            shutdown_event,
+        )
+
+    def start(self) -> None:
+        self.metrics_monitor.start()
+
+    def finish(self) -> None:
+        self.metrics_monitor.finish()
+
+    @classmethod
+    def is_available(cls) -> bool:
+        """Return a new instance of the CPU metrics"""
+        return psutil is not None
+
+    def probe(self) -> dict:
+        """Return a dict of the hardware information"""
+        net_if_addrs = psutil.net_if_addrs()
+
+        return {
+            self.name: {
```

Comment:
I dont think we should include this until we have a reason to include it.
ditto for all the other new probe information items.

You might want to leave this in the code as a disabled item.
or if you can justify why it would be added then it might make sense.

As it stands now.. this is just adding more information into a file that we hope to be getting rid of in the future.

Reply:
done!